tsdb/head_bench_test.go

```diff
@@ -79,6 +86,85 @@ func BenchmarkHeadStripeSeriesCreate_PreCreationFailure(b *testing.B) {
 	}
 }
 
+func BenchmarkHead_WalCommit(b *testing.B) {
+	seriesCounts := []int{100, 1000, 10000}
+	series := genSeries(10000, 10, 0, 0) // Only using the generated labels.
+
+	appendSamples := func(b *testing.B, app storage.Appender, seriesCount int, ts int64) {
+		var err error
+		for i, s := range series[:seriesCount] {
+			var ref storage.SeriesRef
+			if i%2 == 0 {
+				ref, err = app.Append(ref, s.Labels(), ts, float64(ts))
```

Comment:
Could we add a comment explaining why we do %2 == 0? I'm having a hard time understanding it myself 🥲

Reply:
I'll push a comment!
But it's basically if i is even then we append a sample otherwise append histograms.

Reply:
That part I understood, what I couldn't understand is why we do that 🤔. Couldn't we just append them always?

Reply:
I think its to divide the samples between samples and histograms, but I might be wrong.
@krajorama probably will provide a better visition

Reply:
I suggested that to easily add float and histogram samples as well to the mix.